src/main/scala/rocket/CSR.scala

```diff
@@ -1381,6 +1414,13 @@ class CSRFile(
     }
     if (usingUser) {
       when (decoded_addr(CSRs.mcounteren)) { reg_mcounteren := wdata }
+      when (decoded_addr(CSRs.menvcfg))    { reg_menvcfg.write(wdata) }
+    }
+    if (usingSupervisor) {
+      when (decoded_addr(CSRs.senvcfg))    { reg_senvcfg.write(wdata) }
+    }
```

Comment:
https://github.com/chipsalliance/rocket-chip/blob/bd3276d398c50af21812bc054422609491aba3a3/src/main/scala/rocket/CSR.scala#L1309

I believe this is a better place?